dor/providers/process_basic_image.py

```diff
@@ -0,0 +1,212 @@
+)
+from dor.providers.serializers import PreservationEventSerializer
+from dor.providers.utilities import sanitize_basename
+from dor.settings import template_env
+
+
+ACCEPTED_IMAGE_MIMETYPES = [
+    ImageMimetype.JPEG,
+    ImageMimetype.TIFF,
+    ImageMimetype.JP2
+]
+
+
+class FileSetIdentifier:
+
+    def __init__(self, project_id: str, file_name: str):
+        self.project_id = project_id
+        self.file_name = file_name
+        self.basename = sanitize_basename(Path(file_name).stem)
+
+    @property
+    def alternate_identifier(self) -> str:
+        return f"{self.project_id}:{self.basename}"
+
+    @property
+    def uuid(self) -> uuid.UUID:
+        hex_string = hashlib.md5(self.alternate_identifier.encode("UTF-8")).hexdigest()
+        return uuid.UUID(hex=hex_string)
+
+    @property
+    def identifier(self) -> str:
+        return str(self.uuid)
+
+
+def create_preservation_event(
+    type: str,
+    collection_manager_email: str,
+    detail: str = ""
+):
+    event = PreservationEvent(
+        identifier=str(uuid.uuid4()),
+        type=type,
+        datetime=datetime.now(tz=UTC),
+        detail=detail,
+        agent=Agent(
+            role="image_processing",
+            address=collection_manager_email
+        )
+    )
+    return event
+
+
+def convert_metadata_file_info_to_file_metadata(metadata_file_info: MetadataFileInfo) -> FileMetadata:
+    return FileMetadata(
+        id=metadata_file_info.xmlid,
+        use=flatten_use(*metadata_file_info.uses),
+        ref=FileReference(
+            locref=metadata_file_info.locref,
+            mimetype=metadata_file_info.mimetype,
+            mdtype=metadata_file_info.mdtype
+        )
+    )
+
+
+def create_file_set_descriptor_file(
+    resource: PackageResource,
+    descriptor_file_path: Path
+) -> None:
+
+    entity_template = template_env.get_template("preservation_mets.xml")
+    xmldata = entity_template.render(
+        resource=resource,
+        object_identifier=resource.id,
+        create_date=datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ"),
+    )
+    with (descriptor_file_path).open("w") as file:
+        file.write(xmldata)
+
+
+def process_basic_image(
+    file_set_identifier: FileSetIdentifier,
+    input_image_path: Path,
+    output_path: Path,
+    collection_manager_email: str = "example@org.edu",
+    get_technical_metadata: Callable[[Path], TechnicalMetadata] = get_technical_metadata,
+    generate_service_variant: Callable[[Path, Path], None] = generate_service_variant
+) -> bool:
+    identifier = file_set_identifier.identifier
+
+    file_provider = FilesystemFileProvider()
+    file_provider.create_directory(output_path / identifier)
+    file_provider.create_directory(output_path / identifier / "data")
+    file_provider.create_directory(output_path / identifier / "metadata")
+    file_provider.create_directory(output_path / identifier / "descriptor")
+
+    source_file_path = input_image_path
```

Comment:
Perhaps just leave it input_image_path and use source_file_path below instead of new_source_file_path (see other comment as well).

Reply:
Done. https://github.com/mlibrary/dor-py/pull/61/commits/7d77e53ce03beeabbd253fc28271e7c6c7b11909